aten/src/ATen/StorageImpl.h

```diff
@@ -41,29 +42,25 @@ namespace at {
 struct Type;
 
 struct TH_CPP_API StorageImpl : public Retainable {
-
+ public:
   StorageImpl() = delete;
   virtual ~StorageImpl() {};
   StorageImpl(at::ScalarType, ptrdiff_t, at::DataPtr, at::Allocator*, bool);
   StorageImpl(at::ScalarType, ptrdiff_t, at::Allocator*, bool);
-  at::ScalarType scalar_type;
-  at::DataPtr data_ptr;
-  ptrdiff_t size;
-  bool resizable;
-  at::Allocator* allocator;
-  std::unique_ptr<THFinalizer> finalizer;
   StorageImpl(StorageImpl&) = delete;
   StorageImpl(const StorageImpl&) = delete;
-  StorageImpl(StorageImpl&&) = delete;
+  // NB: Don't move ref count!
+  StorageImpl(StorageImpl&& other) = delete;
```

Comment:
FWIW, the default move constructor won't work anyway, because std::atomic is not movable.